crates/router/src/core/admin.rs

```diff
@@ -2796,11 +2796,39 @@ pub async fn retrieve_payment_connector(
             id: merchant_connector_id.clone(),
         })?;
 
-    #[cfg(all(feature = "v2", feature = "merchant_connector_account_v2"))]
-    let mca: domain::MerchantConnectorAccount = {
-        let _ = &merchant_connector_id;
-        todo!()
-    };
+    Ok(service_api::ApplicationResponse::Json(
+        mca.foreign_try_into()?,
+    ))
+}
+
+#[cfg(all(feature = "v2", feature = "merchant_connector_account_v2"))]
+pub async fn retrieve_payment_connector(
```

Comment:
retrieve_connector, in the implement nowhere mentioned it only for payment connector